spec/lib/chess/core/pieces/king_spec.rb

```diff
@@ -0,0 +1,166 @@
+
+RSpec.describe Chess::Core::Pieces::King do
+  subject(:king) { described_class.create_and_occupy(Chess::WHITE_TEAM, cell_e1) }
+
+  let(:cells) do
+    cells = []
+
+    8.times do |row_index|
+      cells << []
+      8.times do |column_index|
+        name = "#{%w[a b c d e f g h][column_index]}#{row_index + 1}"
+        cells[row_index] << Chess::Core::Cell.new(name, Chess::WHITE_TEAM)
+      end
+    end
+
+    cells
+  end
+  let(:cell_e1) { cells[0][4] }
+
+  describe '.create_and_occupy' do
+    it 'occupies the cell' do
+      expect(king.current_cell.occupant).to be(king)
+    end
+  end
+
+  # rubocop:disable RSpec/MultipleMemoizedHelpers
+  describe '#can_move_to?' do
+    let(:cell_d1) { cells[0][3] }
+    let(:cell_d2) { cells[1][3] }
+    let(:cell_e2) { cells[1][4] }
+    let(:cell_f2) { cells[1][5] }
+    let(:cell_f1) { cells[0][5] }
+    let(:cell_c1) { cells[0][2] }
+    let(:cell_e3) { cells[2][4] }
+    let(:cell_e8) { cells[7][4] }
+    let(:cell_h1) { cells[0][7] }
+
+    it 'can move to d1' do
+      expect(king.can_move_to?(cell_d1, cells)).to be(true)
+    end
+
+    it 'can move to d2' do
+      expect(king.can_move_to?(cell_d2, cells)).to be(true)
+    end
+
+    it 'can move to e2' do
+      expect(king.can_move_to?(cell_e2, cells)).to be(true)
+    end
+
+    it 'can move to f2' do
+      expect(king.can_move_to?(cell_f2, cells)).to be(true)
+    end
+
+    it 'can move to f1' do
+      expect(king.can_move_to?(cell_f1, cells)).to be(true)
+    end
+
+    it "can't move to c1" do
+      expect(king.can_move_to?(cell_c1, cells)).to be(false)
+    end
+
+    it "can't move to e3" do
+      expect(king.can_move_to?(cell_e3, cells)).to be(false)
+    end
+
+    it "can't move to e8" do
+      expect(king.can_move_to?(cell_e8, cells)).to be(false)
+    end
+
+    it "can't move to h1" do
+      expect(king.can_move_to?(cell_h1, cells)).to be(false)
+    end
+  end
+  # rubocop:enable RSpec/MultipleMemoizedHelpers
+
+  describe '#move_to' do
+    let(:cell_d1) { cells[0][3] }
+    let(:cell_h1) { cells[0][7] }
+
+    it 'returns target cell' do
+      expect(king.move_to(cell_d1, cells)).to be(cell_d1)
+    end
+
+    it 'returns nil if the move is not possible' do
+      expect(king.move_to(cell_h1, cells)).to be_nil
+    end
+  end
+
+  # rubocop:disable RSpec/MultipleMemoizedHelpers
+  describe '#castle_with' do
+    describe 'queenside' do
+      let(:rook) { Chess::Core::Pieces::Rook.create_and_occupy(Chess::WHITE_TEAM, cell_a1) }
+      let(:cell_a1) { cells[0][0] }
+      let(:cell_c1) { cells[0][2] }
+      let(:cell_f1) { cells[0][5] }
+      let(:cell_d1) { cells[0][3] }
+
+      it 'returns nil if rook was moved' do
+        rook.move_to(cell_c1, cells)
+        expect(king.castle_with(rook, cells)).to be_nil
+      end
+
+      it 'returns nil if king was moved' do
+        king.move_to(cell_f1, cells)
+        king.move_to(cell_e1, cells)
+        expect(king.castle_with(rook, cells)).to be_nil
+      end
```

Comment:
remember what we talked about my preference towards not having anything other than `expect` inside examples? For these scenarios I suggest this:

```suggestion
      describe 'when rook was moved' do
        before do
          rook.move_to(cell_c1, cells)
        end
        
        it 'returns nil' do
          expect(king.castle_with(rook, cells)).to be_nil
        end
      end
      
      describe 'when king was moved' do
        before do
          king.move_to(cell_f1, cells)
          king.move_to(cell_e1, cells)
        end
        
        it 'returns nil' do
          expect(king.castle_with(rook, cells)).to be_nil
        end
      end
```

It's just a suggestion anyways